### PR TITLE
demo/smoke Android build update: use latest cmake with Android Studio

### PR DESCRIPTION
**Pull request by ggfan**
@cnorthrop, @qchong, @bbilodeau, may you kindly have a look?  thanks

### CODE REVIEW

File: demos/smoke/android/build.gradle [hunk 1]
```diff
@@ -4,92 +4,48 @@ buildscript {
...
-                    exclude 'ShellWayland.cpp'
-                }
-            }
-            jniLibs {
-                dependencies {
-                    source.srcDir "${layersDir}"
-                }
-            }
-        }
+    // Need Android NDK-r12+ for pre-built validation layers
```
Comment (cnorthrop) [inline demos/smoke/android/build.gradle]: We want to package this demo with layers built from the repo for use in validation.  I believe that's what was happening above with:
`layersDir = "${demosDir}/../build-android/libs"`

Comment (ggfan) [inline demos/smoke/android/build.gradle]: Done, put back the layers from the local build one, test and it works.

Comment (ggfan) [inline demos/smoke/android/build.gradle]: Yeah, done. it is back now.  thanks

**Comment by cnorthrop** [comment]
cherry-picked to master

**Pull request closed without merging**